server/handlers/validation.go

```diff
@@ -80,9 +80,21 @@ func validateUpdate(cs signed.CryptoService, gun string, updates []storage.MetaU
 			return nil, ErrBadTargets{msg: err.Error()}
 		}
 		repo.SetTargets(targetsRole, t)
+	} else {
+		targetsJSON, err := store.GetCurrent(gun, data.CanonicalTargetsRole)
+		if err != nil {
+			return nil, fmt.Errorf("could not load repo for snapshot generation: %v", err)
```

Comment:
The error isn't necessarily because there's no targets. It could be disrupted connectivity to the database or something like that. At this point in `validateUpdate` the sole purpose of loading the targets.json is to create/validate the snapshot.

So far I've gone with the understanding that the 4 main roles, root, targets, snapshot, and timestamp, must exist both in the root, and their associated json files.